core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -218,23 +218,34 @@ private TableMetadata internalApply() {
     updatedMetaBuilder.removeSnapshots(idsToRemove);
 
     if (cleanExpiredMetadata) {
-      // TODO: Support cleaning expired schema as well.
       Set<Integer> reachableSpecs = Sets.newConcurrentHashSet();
       reachableSpecs.add(base.defaultSpecId());
+      Set<Integer> reachableSchemas = Sets.newConcurrentHashSet();
+      reachableSchemas.add(base.currentSchemaId());
+
       Tasks.foreach(idsToRetain)
           .executeWith(planExecutorService)
           .run(
-              snapshot ->
-                  base.snapshot(snapshot).allManifests(ops.io()).stream()
-                      .map(ManifestFile::partitionSpecId)
-                      .forEach(reachableSpecs::add));
+              snapshotId -> {
+                base.snapshot(snapshotId).allManifests(ops.io()).stream()
```

Comment:
nit: you might want to extract `base.snapshot(snapshotId)` into a separate variable